test/server/cards/02_SHD/upgrades/Foundling.spec.js

```diff
@@ -0,0 +1,32 @@
+describe('Foundling', function () {
+    integration(function () {
+        describe('Foundling', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['foundling'],
+                        groundArena: ['battlefield-marine', 'pyke-sentinel'],
+                    },
+                    player2: {
+                        groundArena: ['snowspeeder']
+                    }
+                });
+            });
+
+            it('should gain mandalorian\'s trait', function () {
+                // check battlefield marine trait
+                expect(this.battlefieldMarine.hasSomeTrait('mandalorian')).toBeFalse();
+
+                // add battlefield marine foundling
+                this.player1.clickCard(this.foundling);
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.snowspeeder, this.pykeSentinel]);
+                this.player1.clickCard(this.battlefieldMarine);
+
+                // check players units traits
+                expect(this.battlefieldMarine.hasSomeTrait('mandalorian')).toBeTrue();
```

Comment:
This is a good start but we always want tests to confirm actual card behavior, which is sometimes tricky because it may involve multiple cards.

In this case the way we would want to confirm that Foundling is working correctly is to use another card that targets Mandalorian cards. Unfortunately, none are implemented yet. If you would be willing to add a simple one like Mandalorian Warrior or Mandalorian Armor in this PR and then use it in this test case, that'd be amazing